packages/releaser/src/ReleaseWorker/ValidateTildaInComposerJsonReleaseWorker.php

```diff
@@ -0,0 +1,147 @@
+    private $isSuccessful = false;
+
+    /**
+     * @var string[]
+     */
+    private $lockVersionAllowedPackages = [];
+
+    /**
+     * @param \Symplify\MonorepoBuilder\FileSystem\ComposerJsonProvider $composerJsonProvider
+     * @param \Symplify\MonorepoBuilder\FileSystem\JsonFileManager $jsonFileManager
+     * @param \Symfony\Component\Console\Style\SymfonyStyle $symfonyStyle
+     * @param string[] $lockVersionAllowedPackages
+     */
+    public function __construct(
+        ComposerJsonProvider $composerJsonProvider,
+        JsonFileManager $jsonFileManager,
+        SymfonyStyle $symfonyStyle,
+        array $lockVersionAllowedPackages
+    ) {
+        $this->composerJsonProvider = $composerJsonProvider;
+        $this->jsonFileManager = $jsonFileManager;
+        $this->symfonyStyle = $symfonyStyle;
+        $this->lockVersionAllowedPackages = $lockVersionAllowedPackages;
+    }
+
+    /**
+     * @return string
+     */
+    public function getDescription(): string
+    {
+        return 'Validate "require" and "require-dev" for missing "^version" formats';
+    }
+
+    /**
+     * Higher first
+     * @return int
+     */
+    public function getPriority(): int
+    {
+        return 920;
+    }
+
+    /**
+     * @param \PharIo\Version\Version $version
+     */
+    public function work(Version $version): void
+    {
+        foreach ($this->composerJsonProvider->getRootAndPackageFileInfos() as $fileInfo) {
+            $jsonContent = $this->jsonFileManager->loadFromFileInfo($fileInfo);
+
+            $this->validateVersions($jsonContent, 'require', $fileInfo);
+            $this->validateVersions($jsonContent, 'require-dev', $fileInfo);
+        }
+
+        if ($this->isSuccessful) {
+            $this->symfonyStyle->success('All versions are in correct format in "require"/"require-dev" sections of each "composer.json');
+        }
+    }
+
+    /**
+     * @param mixed[] $jsonContent
+     * @param string $section
+     * @param \Symfony\Component\Finder\SplFileInfo $splFileInfo
+     */
+    private function validateVersions(array $jsonContent, string $section, SplFileInfo $splFileInfo): void
+    {
+        if (!isset($jsonContent[$section])) {
+            return;
+        }
+
+        foreach ($jsonContent[$section] as $packageName => $version) {
+            if ($this->shouldSkipPackageNameAndVersion($packageName, $version)) {
+                continue;
+            }
+
+            $this->symfonyStyle->warning(sprintf(
+                '"%s" file has invalid version format for "%s": "%s"',
+                $splFileInfo->getPathname(),
+                $packageName,
+                $version
+            ));
+
+            $this->isSuccessful = false;
+        }
+    }
+
+    /**
+     * @param string $packageName
+     * @param string $version
+     * @return bool
+     */
+    private function shouldSkipPackageNameAndVersion(string $packageName, string $version): bool
+    {
+        if ($version === '*') {
```

Comment:
this is ok for `ext-` dependencies (php extensions) only

Reply:
Very good catch :+1:  I didn't realize that